helper/common/common.go

```diff
@@ -375,3 +375,43 @@ func EncodeUint64ToBytes(value uint64) []byte {
 func EncodeBytesToUint64(b []byte) uint64 {
 	return binary.BigEndian.Uint64(b)
 }
+
+// Generic object pool implementation intended to be used in single-threaded
+// manner and avoid synchronization overhead. It could be probably additionally
+// be improved by using circular buffer as oposed to stack.
+type UnsafePool[T any] struct {
```

Comment:
Could we use `sync.Pool` implementation from the Golang? The benefit of this implementation is the ability to use generics, though. WDYT?